rules/react.js

```diff
@@ -104,5 +104,195 @@ module.exports = {
     // Some legacy code heavy use indexes for keys. It is complicated to convert to something else
     // expecially when data is client side derived.
     'react/no-array-index-key': 'warn',
```

Comment:
So this will give us a warning right? Index bugs are really awful to debug and I want at least to see a warning.